pkg/kubelet/kubelet.go

```diff
@@ -61,29 +61,38 @@ type volumeMap map[string]volume.Interface
 
 // New creates a new Kubelet for use in main
 func NewMainKubelet(
-	hn string,
-	dc dockertools.DockerInterface,
-	ec tools.EtcdClient,
-	rd string,
-	ni string,
-	ri time.Duration,
+	hostname string,
+	dockerClient dockertools.DockerInterface,
+	etcdClient tools.EtcdClient,
+	rootDirectory string,
+	networkContainerImage string,
+	resyncInterval time.Duration,
 	pullQPS float32,
 	pullBurst int,
 	minimumGCAge time.Duration,
 	maxContainerCount int,
 	sourcesReady SourcesReadyFn,
 	clusterDomain string,
-	clusterDNS net.IP) *Kubelet {
+	clusterDNS net.IP) (*Kubelet, error) {
+	if resyncInterval <= 0 {
+		return nil, fmt.Errorf("invalid sync frequency %d", resyncInterval)
+	}
+	if minimumGCAge <= 0 {
```

Comment:
To be clear - either decide that 0 is valid and allow it here, or remove the check for == 0